_posts/2021-03-23-solidity-0.8.3-release-announcement.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.3 Release Announcement'
+date: '2021-03-23'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.3](https://github.com/ethereum/solidity/releases/tag/v0.8.3) fixes the [Solidity Optimizer Keccak Caching Bug](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/), which is present in all prior versions of Solidity, and, in addition, includes two improvements to the optimizer which can provide a big gas benefit when writing structs that cover a full storage slot to storage.
+
+## Important Bugfixes
+
+### Solidity Optimizer Keccak Caching Bug
+
+On March 20, 2021, a bug in Solidity’s bytecode optimizer was found by differential fuzzing. The bug is fixed with this version. The bug is present in all prior versions of Solidity and has a severity level of "medium". To learn more about the bug and check if your code is affected by it, please read the corresponding [Security Alert](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/).
+
+## Notable New Features
+
+### Optimizer Improvements
+
+There are two optimizer improvements included in this release, namely improving storage writes for small types and replacing multiplications and divisions with powers of two by shifts. The former will make writing packed structs more gas efficient. In the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:
+
+```solidity
+struct S {
+    address a;
+    uint96 b;
```

Comment:
Don't we need uint136?

Reply:
Just `160 + 96 = 256`. I think this was even the nick johnson's example from twitter.

Reply:
Sorry, my bad. This is correct.